Source/LinqToDB/DataContext.cs

```diff
@@ -386,7 +386,10 @@ IDataContext IDataContext.Clone(bool forNestedQuery)
 		/// <inheritdoc />
 		public Action<EntityCreatedEventArgs>? OnEntityCreated { get; set; }
 
-		void IDisposable.Dispose()
+		/// <summary>
+		/// Closes underlying connection and fires <see cref="OnClosing"/> event (only if connection existed).
+		/// </summary>
+		public virtual void Dispose()
```

Comment:
Per [this document](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose). Also, update the `void Dispose(bool)` to only process if `disposing` is `true` 

```suggestion
		void IDisposable.Dispose()
		{
			Dispose(disposing: true);
			GC.SuppressFinalize(this);
		}
		
		/// <summary>
		/// Closes underlying connection and fires <see cref="OnClosing"/> event (only if connection existed).
		/// </summary>
		protected virtual void Dispose(bool disposing)
```

Reply:
Why? We don't have finalizer in DataContext.
I was always puzzled by this overcomplicated implementation...

Reply:
ah, whatever... let's imagine finalizer in derived class :smile: